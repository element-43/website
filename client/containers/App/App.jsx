import PropTypes from 'prop-types';
import React from 'react';

class App extends React.Component {
    render() {
        return this.props.children && React.cloneElement(this.props.children, { services: 'testy' });
    }
}

App.propTypes = {
    children: PropTypes.node
};

export default App;
