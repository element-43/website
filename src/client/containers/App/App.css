.app {
    composes: primary-font-regular from "../../stylesheets/modules/typography.scss";
    display: flex;
    flex-direction: row;
    overflow: hidden;
    height: 100vh;
    width: 100%;
}
