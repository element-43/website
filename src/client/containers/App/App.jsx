import React, { Component } from 'react';
import CSSModules from 'react-css-modules';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';

// Config.
import { defaults, strings } from '../../../common/index';

// Utilities.
import { getHelmet } from '../../utilities/ApplicationUtil';
import { cssModulesOptions } from '../../utilities/StylesUtil';

// Styles.
import styles from './App.css';

//Components.
import Async from '../../components/Async/Async';
import DefaultLayout from '../../components/DefaultLayout/DefaultLayout';
import GoogleAnalytics from '../../components/GoogleAnalytics/GoogleAnalytics';
import Header from '../../components/Header/Header';
import KonamiCode from '../../components/KonamiCode/KonamiCode';
import MobileHeader from '../../components/MobileHeader/MobileHeader';
import MobileLayout from '../../components/MobileLayout/MobileLayout';
import ScrollTop from '../../components/ScrollToTop/ScrollToTop';
import Terminal from '../../components/Terminal/Terminal';

const menu = [
    {
        title: strings.pages.about.title,
        path: defaults.routes.about
    },
    {
        title: strings.pages.blog.title,
        path: defaults.routes.blog
    },
    {
        title: strings.pages.portfolio.title,
        path: defaults.routes.portfolio
    },
    {
        title: strings.pages.contact.title,
        path: defaults.routes.contact
    }
];

class App extends Component {
    /* eslint-disable max-len */
    render() {
        return (
            <BrowserRouter>
                <div styleName="app">
                    { getHelmet(strings.document.title) }
                    <GoogleAnalytics />
                    <KonamiCode />
                    <ScrollTop />
                    <MobileLayout>
                        <MobileHeader menu={ menu } />
                    </MobileLayout>
                    <DefaultLayout>
                        <Header menu={ menu } />
                    </DefaultLayout>
                    <Switch>
                        <Route
                            exact
                            path='/'
                            component={ props => <Async load={ System.import('../HomePage/HomePage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.about }
                            component={ props => <Async load={ System.import('../AboutPage/AboutPage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.blog }
                            component={ props => <Async load={ System.import('../BlogPage/BlogPage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.contact }
                            component={ props => <Async load={ System.import('../ContactPage/ContactPage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.error }
                            component={ props => <Async load={ System.import('../ErrorPage/ErrorPage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.notFound }
                            component={ props => <Async load={ System.import('../NotFoundPage/NotFoundPage') } { ...props } /> } />
                        <Route
                            path={ defaults.routes.portfolio }
                            component={ props => <Async load={ System.import('../PortfolioPage/PortfolioPage') } { ...props } /> } />
                        <Redirect
                            from="*"
                            to={ defaults.routes.notFound } />
                    </Switch>
                    <Terminal />
                </div>
            </BrowserRouter>
        );
    }
    /* eslint-enable max-len */
}

export default CSSModules(App, styles, cssModulesOptions);
