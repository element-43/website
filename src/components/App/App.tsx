import { ConnectedRouter } from 'connected-react-router';
import { createBrowserHistory, History } from 'history';
import React, { lazy, Suspense } from 'react';
import { Provider } from 'react-redux';
import { Redirect, Route, Switch } from 'react-router-dom';
import { Store } from 'redux';

// Components.
import GlobalStyle from '../GlobalStyle';
import Shell from '../Shell';

// Enums.
import { Routes } from '../../enums';

// Types.
import { ApplicationState } from '../../types';

// Utils.
import { createStore } from '../../utils';

const history: History = createBrowserHistory();
const store: Store<ApplicationState> = createStore(history);

export const App: React.FC = () => (
  <Provider store={store}>
    <GlobalStyle />
    <ConnectedRouter history={history}>
      <Suspense fallback={null}>
        <Shell>
          <Switch>
            <Route
              exact={true}
              path="/"
              component={lazy(() => import('../../pages/Home'))}
            />
            <Route
              exact={true}
              path={Routes.About}
              component={lazy(() => import('../../pages/About'))}
            />
            <Route
              exact={true}
              path={Routes.Contact}
              component={lazy(() => import('../../pages/Contact'))}
            />
            <Route
              exact={true}
              path={Routes.Portfolio}
              component={lazy(() => import('../../pages/Portfolio'))}
            />
            <Redirect from={Routes.Home} to="/" />
            <Redirect from="*" to="/" />
          </Switch>
        </Shell>
      </Suspense>
    </ConnectedRouter>
  </Provider>
);

export default App;
