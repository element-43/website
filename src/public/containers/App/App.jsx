import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';

// Config.
import defaults from '../../../../config/defaults';
import strings from '../../../../config/strings';

// Utilities.
import { getHelmet } from '../../utilities/application.util';

//Components.
import Async from '../../components/Async/Async';
import GoogleAnalytics from '../../components/GoogleAnalytics/GoogleAnalytics';
//import Header from '../../components/Header/Header';
import KonamiCode from '../../components/KonamiCode/KonamiCode';
import ScrollTop from '../../components/ScrollToTop/ScrollToTop';

const App = () => {
    /* eslint-disable max-len */
    return (
        <Router>
            <div className="app">
                { getHelmet(strings.document.title) }
                <GoogleAnalytics />
                <KonamiCode />
                <ScrollTop />
                {/*<Header />*/}
                <Switch>
                    <Route
                        exact
                        path='/'
                        component={ props => <Async load={ System.import('../HomePage/HomePage') } { ...props } /> } />
                    <Route
                        path={ defaults.routes.about }
                        component={ props => <Async load={ System.import('../AboutPage/AboutPage') } { ...props } /> } />
                    <Route
                        path={ defaults.routes.blog }
                        component={ props => <Async load={ System.import('../BlogPage/BlogPage') } { ...props } /> } />
                    <Route
                        path={ defaults.routes.clients }
                        component={ props => <Async load={ System.import('../ClientsPage/ClientsPage') } { ...props } /> } />
                    <Route
                        path={ defaults.routes.error }
                        component={ props => <Async load={ System.import('../ErrorPage/ErrorPage') } { ...props } /> } />
                    <Route
                        path={ defaults.routes.notFound }
                        component={ props => <Async load={ System.import('../NotFoundPage/NotFoundPage') } { ...props } /> } />
                    <Redirect
                        from="*"
                        to={ defaults.routes.notFound } />
                </Switch>
            </div>
        </Router>
    );
    /* eslint-enable max-len */
};

export default App;
