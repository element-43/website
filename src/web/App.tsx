import * as ConnectedReactRouter from 'connected-react-router';
import { createBrowserHistory, History } from 'history';
import * as React from 'react';
import { Provider } from 'react-redux';
import { Redirect, Route, Switch } from 'react-router-dom';
import { Store } from 'redux';

// Components.
import { GlobalStyle } from './components/GlobalStyle';
import { Shell } from './components/Shell';

// Enums.
import { RoutesEnum } from '../common/enums';

// Pages.
import { About } from './pages/About';
import { Contact } from './pages/Contact';
import { Home } from './pages/Home';
import { Portfolio } from './pages/Portfolio';

// Store.
import { ApplicationState, configureStore } from './store';

const history: History = createBrowserHistory();
const Router = ConnectedReactRouter.ConnectedRouter;
const store: Store<ApplicationState> = configureStore(history);

export const App: React.FC = () => (
  <Provider store={store}>
    <GlobalStyle />
    <Router history={history}>
      <Shell>
        <Switch>
          <Route exact={true} path="/" component={Home} />
          <Route exact={true} path={RoutesEnum.About} component={About} />
          <Route exact={true} path={RoutesEnum.Contact} component={Contact} />
          <Route
            exact={true}
            path={RoutesEnum.Portfolio}
            component={Portfolio}
          />
          <Redirect from="*" to="/" />
        </Switch>
      </Shell>
    </Router>
  </Provider>
);
