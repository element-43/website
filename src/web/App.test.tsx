import { shallow } from 'enzyme';
import * as React from 'react';

// Components.
import { App } from './App';

describe('src/App', () => {
  describe('<App /> snapshots', () => {
    it('should match the snapshot with default props', () => {
      expect(shallow(<App />)).toMatchSnapshot();
    });
  });
});
