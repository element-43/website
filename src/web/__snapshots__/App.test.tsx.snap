// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App <App /> snapshots should match the snapshot with default props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BrowserRouter>
        <Connect(PageShell)>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/contact"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Redirect
              from="*"
              push={false}
              to="/"
            />
          </Switch>
        </Connect(PageShell)>
      </BrowserRouter>,
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(PageShell)>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/contact"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Redirect
              from="*"
              push={false}
              to="/"
            />
          </Switch>
        </Connect(PageShell)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/about"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/contact"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Redirect
              from="*"
              push={false}
              to="/"
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/contact"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />,
              <Redirect
                from="*"
                push={false}
                to="/"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/contact",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/portfolio",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "from": "*",
                "push": false,
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BrowserRouter>
          <Connect(PageShell)>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/contact"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />
              <Redirect
                from="*"
                push={false}
                to="/"
              />
            </Switch>
          </Connect(PageShell)>
        </BrowserRouter>,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(PageShell)>
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/contact"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />
              <Redirect
                from="*"
                push={false}
                to="/"
              />
            </Switch>
          </Connect(PageShell)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/about"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/contact"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />
              <Redirect
                from="*"
                push={false}
                to="/"
              />
            </Switch>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/about"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/contact"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/portfolio"
                />,
                <Redirect
                  from="*"
                  push={false}
                  to="/"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/about",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/contact",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/portfolio",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "from": "*",
                  "push": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
